tensorflow/tools/dist_test/python/mnist_replica.py

```diff
@@ -137,128 +137,133 @@ def main(unused_argv):
-
-    train_step = opt.minimize(cross_entropy,
-                              global_step=global_step)
-
-    if FLAGS.sync_replicas and is_chief:
-      # Initial token and chief queue runners required by the sync_replicas mode
-      chief_queue_runner = opt.get_chief_queue_runner()
-      init_tokens_op = opt.get_init_tokens_op()
-
-    init_op = tf.initialize_all_variables()
-    train_dir = tempfile.mkdtemp()
-    sv = tf.train.Supervisor(is_chief=is_chief,
-                             logdir=train_dir,
-                             init_op=init_op,
-                             recovery_wait_secs=1,
-                             global_step=global_step)
-
-    sess_config = tf.ConfigProto(
-        allow_soft_placement=True,
-        log_device_placement=True,
-        device_filters=["/job:ps", "/job:worker/task:%d" % FLAGS.worker_index])
-
-    # The chief worker (worker_index==0) session will prepare the session,
-    # while the remaining workers will wait for the preparation to complete.
-    if is_chief:
-      print("Worker %d: Initializing session..." % FLAGS.worker_index)
-    else:
-      print("Worker %d: Waiting for session to be initialized..." %
-            FLAGS.worker_index)
-
-    sess = sv.prepare_or_wait_for_session(FLAGS.worker_grpc_url,
-                                          config=sess_config)
-
-    print("Worker %d: Session initialization complete." % FLAGS.worker_index)
-
-    if FLAGS.sync_replicas and is_chief:
-      # Chief worker will start the chief queue runner and call the init op
-      print("Starting chief queue runner and running init_tokens_op")
-      sv.start_queue_runners(sess, [chief_queue_runner])
-      sess.run(init_tokens_op)
-
-    # Perform training
-    time_begin = time.time()
-    print("Training begins @ %f" % time_begin)
-
-    local_step = 0
-    while True:
-      # Training feed
-      batch_xs, batch_ys = mnist.train.next_batch(FLAGS.batch_size)
-      train_feed = {x: batch_xs,
-                    y_: batch_ys}
-
-      _, step = sess.run([train_step, global_step], feed_dict=train_feed)
-      local_step += 1
-
-      now = time.time()
-      print("%f: Worker %d: training step %d done (global step: %d)" %
-            (now, FLAGS.worker_index, local_step, step))
-
-      if step >= FLAGS.train_steps:
-        break
-
-    time_end = time.time()
-    print("Training ends @ %f" % time_end)
-    training_time = time_end - time_begin
-    print("Training elapsed time: %f s" % training_time)
-
-    # Validation feed
-    val_feed = {x: mnist.validation.images,
-                y_: mnist.validation.labels}
-    val_xent = sess.run(cross_entropy, feed_dict=val_feed)
-    print("After %d training step(s), validation cross entropy = %g" %
-          (FLAGS.train_steps, val_xent))
+      if FLAGS.replicas_to_aggregate is None:
+        replicas_to_aggregate = FLAGS.num_workers
+      else:
+        replicas_to_aggregate = FLAGS.replicas_to_aggregate
```

Comment:
Move this `if` statement to the same block where the `tf.train.SyncReplicasOptimizer` is constructed.